[DOCS] Updating approle docs and token partial to include batch token prefer… (#26490)
hashicorp/vault: website/content/partials/tokenstorefields.mdx

  unless changed will be `service` tokens). For token store roles, there are two
  additional possibilities: `default-service` and `default-batch` which specify
  the type to return unless the client requests a different type at generation

>   time. If using machine based authentication it is recommended to use `batch` 

Reviewer: ```suggestion
  time. For machine based authentication cases, you should use `batch` 
```
